docs/toggleable-passes.md

```diff
@@ -0,0 +1,126 @@
+# Passes
+
+|  flag          | Default  | What it does? |
+|----------------|----------|---------------|
+| `-Oall`        | Disabled | Enables all compiler passes |
+| `-Ono-all`     | Disabled | Disables all compiler passes |
+| `-Oeta` `-Ono-eta` | Disabled | [eta-reduction](#eta-reduction) |
+| `-Oref-to-ref` `-Ono-ref-to-ref` | Disabled | [ref-to-ref](#ref-to-ref) |
+| `-Oprune` `-Ono-prune` | Disabled | [definition-prununing](#definition-pruning) |
+| `-Opre-reduce` `-Ono-pre-reduce` | Disabled | [pre-reduce](#pre-reduce) |
+| `-Osupercombinators` `-Ono-supercombinators` | Enabled  | [supercombinators](#supercombinators) |
+| `-Osimplify-main` `-Ono-simplify-main` | Disabled | [simplify-main](#simplify-main) |
+
+## Eta-reduction
+
+Enables or disables Eta Reduction for compiler generated functions.
```

Comment:
Generated functions are always created eta-reduced. This pass is to enable eta-reduction for user-provided functions.